overlord/hookstate/hookmgr_test.go

```diff
@@ -265,3 +358,64 @@ func (h *mockHandler) Error(err error) error {
 	}
 	return nil
 }
+
+type mockProcessRunner struct {
```

Comment:
Realted to the point about `ProcessRunner` itself, this is a ton of logic which can be easily achieved by `testutil.MockCommand`, with the advantage that you'll be testing it all the way to real process interactions.